crates/trie/sparse/src/trie.rs

```diff
@@ -31,13 +45,36 @@ impl TrieMasks {
     }
 }
 
-/// Inner representation of the sparse trie.
-/// Sparse trie is blind by default until nodes are revealed.
-#[derive(PartialEq, Eq)]
+/// A sparse trie that is either in a "blind" state (nodes are hidden and only represented
+/// by their hashes) or in a "revealed" state (nodes have been loaded and can be updated).
+///
+/// In blind mode the trie does not contain any decoded node data, which saves memory but
+/// prevents direct access to node contents. The revealed mode stores decoded nodes along
+/// with additional information such as values, allowing direct manipulation.
+///
+/// The sparse trie design is optimised for:
+/// 1. Memory efficiency - only needed nodes are loaded into memory
+/// 2. Update tracking - changes to the trie structure can be tracked and selectively persisted
+/// 3. Incremental operations - nodes can be revealed as needed without loading the entire trie
```

Comment:
1 - all revealed nodes are loaded in memory, so this is not technically correct I think
3 - this is the "sparse" portion of the trie, worth mentioning it i think

Reply:
For 1. I was thinking about memory efficiency of a sparse trie vs a normal trie where you can't have subtrees of the trie being blinded and thus not loaded into memory
3. Makes sense